evm_trace/gas.py

```diff
@@ -0,0 +1,49 @@
+import copy
+from typing import Any, Dict, List
+
+from evm_trace.base import CallTreeNode
+
+GasReport = Dict[Any, Dict[Any, List[int]]]
+
+
+def get_gas_report(calltree: CallTreeNode) -> GasReport:
+    """
+    Extracts a gas report object from a :class:`~evm_trace.base.CallTreeNode`.
+
+    Args:
+        calltree (:class:`~evm_trace.base.CallTreeNode`): call tree used for gas report.
+
+    Returns:
+        :class:`~evm_trace.gas.Report`: Gas report structure from a call tree.
+    """
+    report = {
+        calltree.address: {calltree.calldata[:4]: [calltree.gas_cost] if calltree.gas_cost else []}
+    }
+
+    report = merge_reports([report, *map(get_gas_report, calltree.calls)])
+
+    return report
+
+
+def merge_reports(reports: List[GasReport]) -> GasReport:
```

Comment:
Would make me so happy to do this instead lol
```suggestion
def merge_reports(*reports: List[GasReport]) -> GasReport:
```